File: Java/leetcode/medium/RotateArray.java
package Java.leetcode.medium;

import java.util.Arrays;

public class RotateArray {
    public static void main(String[] args) {
        int[] nums = {1,2,3,4,5,6,7};
        rotateArray(nums, 3);
        System.out.println(Arrays.toString(nums));
    }

    public static void rotateArray(int[] nums, int k){
        k = k % nums.length;
       reverse(nums, 0, nums.length -1 );
       reverse(nums, 0, k-1);
       reverse(nums, k, nums.length - 1);
    }

    public static void reverse(int[] nums, int left, int right){

        while(left < right){
            int temp = nums[left];
            nums[left] = nums[right];
            nums[right] = temp;
            left++;
            right--;
        }
    }
}

File: Javascript/leetcode/medium/RotateArray.js
const RotateArray = (nums, k) => {
  k = k % nums.length
  const spliced = nums.splice(nums.length - k, k)
  nums.unshift(...spliced)
  console.log(nums)
}

//USING REVERSE
const rotateArray = (nums, k) => {
  k = k % nums.length
  reverse(nums, 0, nums.length - 1)
  reverse(nums, 0, k - 1)
  reverse(nums, k, nums.length - 1)
}

const reverse = (nums, start, end) => {
  while (start < end) {
    let temp = nums[start]
    nums[start] = nums[end]
    nums[end] = temp
    start++
    end--
  }
}

console.log(rotateArray([1, 2, 3, 4, 5, 6, 7], 3))
